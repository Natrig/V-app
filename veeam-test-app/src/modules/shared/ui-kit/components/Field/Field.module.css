.root {
    display: flex;
    justify-content: space-between;
    padding-bottom: 10px;
}

.root > label {
    margin-right: 15px;
}

.root > input, .root > textarea {
    width: 200px;
}
