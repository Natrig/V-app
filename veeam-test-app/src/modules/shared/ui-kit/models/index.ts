export enum FieldType {
  Text = 'text',
  Number = 'number',
  Checkbox = 'checkbox',
  Date = 'date',
  Radio = 'radio',
  Textarea = 'textarea',
}

export enum ButtonType {
  Cancel = 'cancel',
  Confirm = 'confirm',
}

export interface CommonField {
  label: string;
  name: string;
  type: FieldType;
  value?: string | number;
  checked?: boolean;
  optionValues?: string[];
}

export interface CommonButton {
  label: string;
  type: ButtonType;
}

export type CommonValue = string | number | boolean | undefined;
