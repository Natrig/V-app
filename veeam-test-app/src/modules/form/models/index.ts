import { CommonField, CommonButton, CommonValue } from '../../shared/ui-kit/models';

export interface FormType {
  controls: CommonButton[];
  fields: CommonField[];
  title: string;
}

export interface FormData {
  name: string;
  value: CommonValue;
}
