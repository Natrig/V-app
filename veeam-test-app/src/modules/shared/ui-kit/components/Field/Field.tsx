import React, { FC } from 'react';
import classNames from 'classnames';

import { CommonField, FieldType, CommonValue } from 'modules/shared/ui-kit/models';
import { Input } from 'modules/shared/ui-kit/components/Input';
import { Textarea } from 'modules/shared/ui-kit/components/Textarea';

import styles from './Field.module.css';

export interface FieldProps {
  field: CommonField;
  onChange: (value: CommonValue, field: CommonField) => void;
  className?: string;
}

export const Field: FC<FieldProps> = ({ field, onChange, className }) => {
  const { type } = field;
  const fieldClassname = classNames([styles.root, className]);

  if (type === FieldType.Textarea) {
    return <Textarea onChange={onChange} field={field} className={fieldClassname} />;
  }

  return <Input onChange={onChange} field={field} className={fieldClassname} />;
};
