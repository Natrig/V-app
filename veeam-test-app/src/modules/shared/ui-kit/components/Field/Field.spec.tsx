import React from 'react';
import { render } from '@testing-library/react';

import { Field } from './Field';
import { CommonField, CommonValue, FieldType } from '../../models';

const defaultProps = {
  onChange: (value: CommonValue, field: CommonField) => {
    return 'test';
  },
};

const defaultFieldProps = {
  label: 'test',
  name: 'test',
};

describe('Field', () => {
  it('should render Text input', () => {
    const props = {
      ...defaultProps,

      field: {
        ...defaultFieldProps,
        type: FieldType.Text,
      },
    };

    const { container } = render(<Field {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render Number input', () => {
    const props = {
      ...defaultProps,

      field: {
        ...defaultFieldProps,
        type: FieldType.Number,
      },
    };

    const { container } = render(<Field {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render Checkbox input', () => {
    const props = {
      ...defaultProps,

      field: {
        ...defaultFieldProps,
        type: FieldType.Checkbox,
        checked: true,
      },
    };

    const { container } = render(<Field {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render Textarea input', () => {
    const props = {
      ...defaultProps,

      field: {
        ...defaultFieldProps,
        type: FieldType.Textarea,
      },
    };

    const { container } = render(<Field {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render RadioGroup input', () => {
    const props = {
      ...defaultProps,

      field: {
        ...defaultFieldProps,
        type: FieldType.Radio,
        optionValues: ['test1', 'test2', 'test3'],
      },
    };

    const { container } = render(<Field {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
