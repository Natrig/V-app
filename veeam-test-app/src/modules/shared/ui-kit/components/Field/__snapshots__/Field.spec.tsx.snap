// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field should render Checkbox input 1`] = `
<div
  class="root"
>
  <label
    for="test"
  >
    test
  </label>
  <input
    id="test"
    type="checkbox"
  />
</div>
`;

exports[`Field should render Number input 1`] = `
<div
  class="root"
>
  <label
    for="test"
  >
    test
  </label>
  <input
    id="test"
    type="number"
    value=""
  />
</div>
`;

exports[`Field should render RadioGroup input 1`] = `
<div
  class="root"
>
  <label
    for="test"
  >
    test
  </label>
  <div>
    <span>
      <input
        id="test"
        type="radio"
        value="test1"
      />
      test1
    </span>
    <span>
      <input
        id="test"
        type="radio"
        value="test2"
      />
      test2
    </span>
    <span>
      <input
        id="test"
        type="radio"
        value="test3"
      />
      test3
    </span>
  </div>
</div>
`;

exports[`Field should render Text input 1`] = `
<div
  class="root"
>
  <label
    for="test"
  >
    test
  </label>
  <input
    id="test"
    type="text"
    value=""
  />
</div>
`;

exports[`Field should render Textarea input 1`] = `
<div
  class="root"
>
  <label
    for="test"
  >
    test
  </label>
  <textarea
    id="test"
    title="test"
  />
</div>
`;
